mem0/memory/main.py

```diff
@@ -1141,13 +1141,12 @@ async def _add_to_vector_store(
             response = remove_code_blocks(response)
             new_retrieved_facts = json.loads(response)["facts"]
         except Exception as e:
+            logging.error(f"Error in new_retrieved_facts: {e}")
             new_retrieved_facts = []
 
         if not new_retrieved_facts:
             logger.info("No new facts retrieved from input. Skipping memory update LLM call.")
             return []
-            logging.error(f"Error in new_retrieved_facts: {e}")
-            new_retrieved_facts = []
```

Comment:
Note: code was made unreachable in PR https://github.com/mem0ai/mem0/pull/2744, it means if the response isn't valid JSON or the JSON malformed, ie. no `facts` key, the error will no longer be logged.